apps/dc_tools/odc/apps/dc_tools/index_from_tar.py

```diff
@@ -1,13 +1,135 @@
+    constellation = properties['constellation']
+
+    if constellation in KNOWN_CONSTELLATIONS:
+        if constellation == 'sentinel-2':
+            product_label = properties['sentinel:product_id']
+            product_name = 's2_l2a'
+            region_code = '{}{}{}'.format(
+                str(properties['proj:epsg'])[-2:],
+                properties['sentinel:latitude_band'],
+                properties['sentinel:grid_square']
+            )
+
+    return product_label, product_name, region_code
+
+
+def _get_stac_bands(item, default_grid='g10.0m'):
+    bands = {}
+
+    grids = {}
+
+    assets = item['assets']
+
+    for asset_name, asset in assets.items():
+        # Ignore items that are not actual COGs
+        if asset['type'] not in ['image/tiff; application=geotiff; profile=cloud-optimized']:
+            continue
+
+        transform = asset['proj:transform']
+        grid = f'g{transform[0]}m'
+
+        if grid not in grids:
+            grids[grid] = {
+                'shape': asset['proj:shape'],
+                'transform': asset['proj:transform']
+            }
+
+        band_info = {
+            'path': os.path.split((asset['href']))[1]
+        }
+
+        if grid != default_grid:
+            band_info['grid'] = grid
+
+        bands[asset_name] = band_info
+
+    grids['default'] = grids[default_grid]
+    del grids[default_grid]
+
+    return bands, grids
+
+
+def geographic_to_projected(geometry, target_srs):
+    """ Transform from WGS84 to the target projection, assuming Lon, Lat order
+    """
+    transformer = Transformer.from_crs(4326, target_srs, always_xy=True)
+    geometry['coordinates'][0] = list(transformer.itransform(
+        geometry['coordinates'][0]))
+
+    return geometry
+
+
+def stac_transform(input_stac):
+    """ Takes in a raw STAC 1.0 dictionary and returns an ODC dictionary
+    """
+
+    product_label, product_name, region_code = _stac_product_lookup(input_stac)
+
+    deterministic_uuid = str(odc_uuid("sentinel-2_stac_process", "1.0.0", [product_label]))
+
+    bands, grids = _get_stac_bands(input_stac)
+
+    properties = input_stac['properties']
+
+    stac_odc = {
+        '$schema': 'https://schemas.opendatacube.org/dataset',
+        'id': deterministic_uuid,
+        'crs': f"epsg:{properties['proj:epsg']}",
+        'geometry': geographic_to_projected(
+            input_stac['geometry'],
+            properties['proj:epsg']
+        ),
+        'grids': grids,
+        'product': {
+            'name': product_name.lower()
+        },
+        'label': product_label,
+        'properties': {
+            'datetime': properties['datetime'].replace("000+00:00", "Z"),
+            'odc:processing_datetime': properties['datetime'].replace("000+00:00", "Z"),
+            'eo:cloud_cover': properties['eo:cloud_cover'],
+            'eo:gsd': properties['gsd'],
+            'eo:instrument': properties['instruments'][0],
+            'eo:platform': properties['platform'],
```

Comment:
how come we prefix `eo:` for `gsd` and `platform`, has this been moved from extension to stac-proper? if yes should we update eo3 metadata then?